haxepunk/utils/DrawContext.hx

```diff
@@ -384,11 +400,17 @@ class DrawContext
 
 	/** @private Helper function to grab a DrawCommand object from the current scene */
 	@:access(haxepunk.graphics.hardware.SceneRenderer)
-	inline function begin()
+	@:access(haxepunk.graphics.hardware.DrawCommand)
+	function begin()
 	{
-		if (shader == null) shader = new ColorShader();
-		var scene = (this.scene == null) ? (HXP.renderingScene == null ? HXP.scene : HXP.renderingScene) : this.scene;
-		command = scene.renderer.batch.getDrawCommand(null, shader, smooth, blend, null);
+		var scene = this.scene.or(HXP.engine.activeScene.ensure());
+		if (needsCommand || scene != lastScene || command._next != null)
```

Comment:
Something doesn't work with this check. It reduces the number of `getDrawCommand` calls but occasionally causes Draw commands to not render. I'm going to have to look into this before merging this code.